akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -427,6 +428,7 @@ private[remote] class Association(
 
   private def attachStreamRestart(streamName: String, streamCompleted: Future[Done], restart: Throwable ⇒ Unit): Unit = {
     implicit val ec = materializer.executionContext
+    updateStreamCompletion(streamName, streamCompleted)
```

Comment:
In https://github.com/akka/akka/pull/21356 I have added the queueIndex as parameter here so then we could use that as key. The purpose of the `streamName` was only for logging.